enf-reducer/cut-unnecessary-or.metta

```diff
@@ -0,0 +1,19 @@
+ ; Cut unnecessary OR, DOESN'T process sub expressions
+ ; To be used with iterator
+( = (orCut $expresion)
+    (if  (== $expresion ()) ()
+        ( let  $type (get-metatype $expresion)
+             ; Check if what is passed in is a symbol/Literal and not an expression
+            (if (== $type Symbol)
+                $expresion
+                (let* (
+                        ($head (car-atom $expresion))
+                        ($tail (cdr-atom $expresion))
+                    ) (if (== $head OR)
+                    (let* (
+                            ($subhead (car-atom $tail))
+                            ($subtail (cdr-atom $tail))
+                        ) (if (== $subtail ()) $subhead (cons-atom $head $tail)))
```

Comment:
This isn't a correct implementation as described on the paper. The function should be returning the `guardSet` and the `children` of the grandchild expression instead of returning the whole grandchild itself as you did here. Make sure to update your code as such.

Reply:
And there is no need to use `(cons-atom $head $tail)` here. You can just return the initial `$expression`

Reply:
> This isn't a correct implementation as described on the paper. The function should be returning the `guardSet` and the `children` of the grandchild expression instead of returning the whole grandchild itself as you did here. Make sure to update your code as such.

I checked the description, and I am not sure I understand what you mean, maybe in the form of input and expected output.

Reply:
@BrookFeleke In the paper, if you understood the description correctly. i.e., if the `POA` is for instance the `OR` node in the expression `(AND B (OR (AND (NOT D) C)))` the cut unnecessary or transformation will returns the expression `(AND B (NOT D) C)`. But your implementation is returning `(AND B (AND (NOT D) C)))`. Which isn't correct.